src/replication.c

```diff
@@ -1154,12 +1220,27 @@ update_replica_entry(spec_t *spec, replica_t *replica, int iofd)
 		close(iofd);
 		return -1;
 	}
+
+	if (can_replica_connected(spec, replica) == false) {
```

Comment:
Add a comment above saying
'After `can_replica_connect` check above, there might be any non_quorum replicas became healthy. If (healthy+degraded) reaches replication_factor, this replica addition won't be required. So, same check required here with the spec lock'